snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/SnomedRf2ExportRequest.java

```diff
@@ -484,7 +486,15 @@ private List<String> computeBranchesToExport(final String referenceBranch, final
 				}
 				break;
 			case SNAPSHOT:
-				branchesToExport.add(referenceBranch);
+				// in case we are exporting published content only use the version branch instead of the reference branch
+				if (!this.includePreReleaseContent) {
+					if (versionsToExport.isEmpty()) {
+						throw new BadRequestException("Snapshot export without unpublished components requires at least one version.");
+					}
+					branchesToExport.add(versionsToExport.first().getBranchPath());
```

Comment:
`versionsToExport` is a `TreeSet` with an ascending effective time sort order. I think we need the element with the greatest value here, not the smallest.

To confirm, add at least one more version to the test case that makes a different change on the component.

Reply:
Yes, you are right I thought that is a TreeSet but the versions are in reverse order because of the ordering we use when fetching the versions. The Ordering at the top of the class, unfortunately, orders them by chronological effective time order. To reflect this, I fixed the issue, renamed the variable, and updated the test case with an additional version and component.